src/librustdoc/html/format.rs

```diff
@@ -1431,20 +1431,16 @@ impl clean::FnDecl {
         cx: &Context<'_>,
     ) -> fmt::Result {
         let amp = if f.alternate() { "&" } else { "&amp;" };
-
+        
         write!(f, "(")?;
         if let Some(n) = line_wrapping_indent
             && !self.inputs.values.is_empty()
         {
             write!(f, "\n{}", Indent(n + 4))?;
         }
+
+        let last_input_index = self.inputs.values.len() - 1;
```

Comment:
this crashes in debug builds when `self.inputs.values.is_empty()` due to negative overflow (it wraps around in release)

Reply:
I'm sorry, but i can't understand this comment, `last_input_index` would be -1 if the inputs vector is empty, but since is only used to compare indexes inside the for loop i can't see the problem. 
There is something i'm not noticing?

Reply:
> since [it's] only used to compare indexes inside the for loop i can't see the problem

It's not inside a for-loop, therefore it can indeed trigger. Otherwise you'd be absolutely right.